spec/unit/actions/synchronize_users_spec.rb

```diff
@@ -100,16 +107,87 @@
       end
     end
 
-    context 'when using mode sync_domain_client_workspaces' do
+    context 'when using multiple_projects modes' do
+      let(:segment) { double('segment') }
+      let(:segment_uri) { 'segment_uri' }
+      let(:message_for_project) { :import_users }
+      let(:params_stub) do
+        {
+          GDC_GD_CLIENT: client,
+          input_source: {},
+          domain: 'bar',
+          gdc_logger: logger,
+          segments: [segment]
+        }
+      end
+
+      before do
+        allow(segment).to receive(:uri).and_return(segment_uri)
+        allow(segment).to receive(:segment_id).and_return('123')
+        allow(segment).to receive(:development_pid).and_return('456')
+        allow(segment).to receive(:driver).and_return('vertica')
+        allow(segment).to receive(:production_tags).and_return([])
+        allow(segment).to receive(:master_name).and_return('charles')
+        allow(organization).to receive(:segment_uri).and_return(segment_uri)
+        allow(organization).to receive(:project).and_return(project)
+        allow(organization).to receive(:client_id).and_return('123456789')
+        allow(project).to receive(:deleted?).and_return(false)
+      end
+
+      context 'sync_multiple_projects_based_on_custom_id mode' do
+        let(:params) do
+          GoodData::LCM2.convert_to_smart_hash(params_stub.merge(sync_mode: 'sync_multiple_projects_based_on_custom_id',
+                                                                              multiple_projects_column: 'client_id'))
+        end
+        before do
+          allow(File).to receive(:open).and_return("client_id\n123456789")
+          allow(domain).to receive(:clients).with(:all, nil).and_return([organization, organization_not_in_segment])
+          allow(domain).to receive(:clients).with('123456789', nil).and_return(organization)
+        end
+
+        it_behaves_like 'a user action filtering segments'
+      end
+
+      context 'sync_multiple_projects_based_on_pid mode' do
```

Comment:
warning: Similar code found in synchronize_user_filters_spec.rb:87, synchronize_users_spec.rb:137, synchronize_users_spec.rb:151, synchronize_users_spec.rb:165 (mass = 204) (flay)